.github/workflows/run-acceptance-tests.yml

```diff
@@ -140,6 +140,15 @@ jobs:
       with:
         name: pulumi-${{ env.PROVIDER }}-provider.tar.gz
         path: ${{ github.workspace }}/bin/provider.tar.gz
+    - name: Configure AWS Credentials
```

Comment:
This file is auto generated, we'll need to change it here: https://github.com/pulumi/ci-mgmt

Reply:
I can do this - I guess my first question is whether I _should_ be configuring credentials here and why I'm the first one to need this. Should I be using some other place for end-2-end tests?

Reply:
I think it makes sense to do that for now. They already include upgrade tests and while they do not use any invokes right now, they might in the future and those need creds: https://github.com/pulumi/pulumi-aws-native/blob/11ac7b3942424e3c4863677b2a6ead0de2ab37bc/provider/pkg/provider/provider_2e2_test.go#L19-L31

(Just saw that the upgrade tests are missing `-short` flag, I'll add that separately)

Long term I'd like us to have a more standard test setup where we have a single go.mod on the root and then separate directories for tests in order to keep slower-running integration tests separate from unit tests. This would enable us to run unit tests without needing to remember `-short` or build tags. But that's all unrelated to your changes.

Reply:
Working on CI in https://github.com/pulumi/ci-mgmt/pull/1239 but also discussing offline what best approach could be

Reply:
https://github.com/pulumi/ci-mgmt/pull/1239 is typically how we'd unblock this, but we're trying to do a couple things right now in the opposite direction:
- we want to reduce the number of exceptions for native providers in order to make it easier to consolidate them with bridged workflows
- and we want to make CI tests more reproducible locally, which means reducing the reliance on CI-specific setup steps.

For those reasons I'll suggest an alternative approach. Basically just perform the auth in-process as part of your test's setup. Actually, before you do that maybe just plumb the environment variables through, since the provider should login for you? So the test would get an environmnet

```go
Env: []string{"AWS_ROLE_SESSION_NAME=" + os.Getenv("PROVIDER"), "AWS_ROLE_TO_ASSUME=" + os.Getenv("AWS_CI_ROLE_ARN"), ...}
```
Otherwise, if you do need to perform the auth, something like this pseudocode:

```go
cfg := aws.Config{
  Credentials: credentials.NewStaticCredentialsProvider(accessKey, secretKey, ""),
  Region: region,
}

c := sts.NewFromConfig(cfg)
creds, err := c.AssumeRole(ctx, &sts.AssumeRoleInput{
  RoleArn:         aws.String(os.Getenv("AWS_CI_ROLE_ARN")),
  RoleSessionName: aws.String(os.Getenv("AWS_ROLE_SESSION_NAME")),
})
require.NoError(t, err)

// Set environment with creds
```


Then pass those environment variables to the e2e test via `Env:`, or (worse) if you're lazy use `os.Setenv`.

If `provider/pkg/provider/provider_2e2_test.go` is the only place where you need these credentials, then I suggest defining a helper there like `loginToAWS(t)`. You can put it behind a [`sync.Once`](https://pkg.go.dev/sync) to prevent logging in more than you need to.

Give it a try, and if it turns into a rabbit hole we can merge the ci-mgmt PR and clean it up later.

Reply:
I can give this a try... This would make it unusable in a local dev environment unless one replicates all env vars, right?

Reply:
I'd personally vote for the `ci-mgmt` option because it doesn't mess with the local dev setup. The changes to `ci-mgmt` are not that big IMO and the aws-native test structure will have to change for the consolidation anyways

Reply:
I think I prefer `ci-mgmt` too... I un-drafted that PR, happy to ship it and remove the CI change here after it percolates.